### PR TITLE
#1035: Github action to update ide-urls

### PR DESCRIPTION
**Pull request by alfeilex**
This PR creates a GitHub Action to update urls in https://github.com/devonfw/ide-urls repository.

Related issue: #1035

### CODE REVIEW

File: .github/workflows/update-urls.yml
Comment (alfeilex) [inline .github/workflows/update-urls.yml:18]: This token needs repo scope permissions. If there is already a token in `devonfw/ide` for this, we can update.

**Comment by alfeilex** [comment]
See results in forked ide-urls repo:

https://github.com/alfeilex/ide-urls

**Review by hohwille — COMMENTED** [review]
@alfeilex thanks for your PR and this workflow. Nice work 👍 
As the Java code executed from the workflow is not yet merged, we have to wait for this before we can merge and use this workflow. Thanks also for the hint about the token.

**Review by tobka777 — APPROVED** [review]
Looks good. We should merge it and then test if it works incl. the tokens.

File: .github/workflows/update-urls.yml
Comment (hohwille) [inline .github/workflows/update-urls.yml:31]: as discussed in PR #1093 we need to solve the push problem as this will also not work as is.

**Comment by hohwille** [comment]
#1086 needs to be implemented and merged before merging this PR to avoid clutter in git history.

File: .github/workflows/update-urls.yml
Comment (hohwille) [reply .github/workflows/update-urls.yml:18]: We do not have such token yet. Do you have a link to any github instructions about this?
I know how to create secrets but should I add my personal github access token there or what value should this have?

Comment (alfeilex) [reply .github/workflows/update-urls.yml:18]: https://docs.github.com/de/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#erstellen-eines-personal-access-token-classic

I added my own personal access token there. That was the only way I could find to make it work.

Comment (hohwille) [reply .github/workflows/update-urls.yml:18]: OK, I have created such token as fine-grained org token (was more complicated than I expected) and added it as according secret to the devonfw github org (even though so far the token is only limited to ide specific repos). Lets first test this approach. Also the token requires an expiry so the github action will stop working on 2.4.2025.

File: .github/workflows/update-urls.yml
Comment (hohwille) [reply .github/workflows/update-urls.yml:31]: Lets give it a try with the push token...

**Pull request merged**
